storage/storage_image.go

```diff
@@ -1197,21 +1197,15 @@ func (s *storageImageDestination) Commit(ctx context.Context, unparsedToplevel t
 		}
 		logrus.Debugf("saved image metadata %q", string(metadata))
 	}
-	// Set the reference's name on the image.  We don't need to worry about avoiding duplicate
-	// values because SetNames() will deduplicate the list that we pass to it.
-	if name := s.imageRef.DockerReference(); len(oldNames) > 0 || name != nil {
+	// Adds the reference's name on the image.  We don't need to worry about avoiding duplicate
+	// values because AddNames() will deduplicate the list that we pass to it.
+	if name := s.imageRef.DockerReference(); name != nil {
 		names := []string{}
-		if name != nil {
-			names = append(names, name.String())
+		names = append(names, name.String())
+		if err := s.imageRef.transport.store.AddNames(img.ID, names); err != nil {
```

Comment:
With the two lines above deleted.
```suggestion
		if err := s.imageRef.transport.store.AddNames(img.ID, string[]{name.String()}); err != nil {
```